src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/InternationalPostalService.java

```diff
@@ -0,0 +1,216 @@
+     *
+     * @param caseData The FinremCaseData to validate.
+     * @return A list of error messages. An empty list indicates no errors were found.
+     */
+    public List<String> validate(FinremCaseData caseData) {
+        ContactDetailsWrapper wrapper = caseData.getContactDetailsWrapper();
+        YesOrNo applicantResideOutsideUK = wrapper.getApplicantResideOutsideUK();
+        List<String> errors = new ArrayList<>();
+        if (ObjectUtils.isNotEmpty(applicantResideOutsideUK)
+            && applicantResideOutsideUK.equals(YesOrNo.YES)
+            && wrapper.getApplicantAddress() != null
+            && ObjectUtils.isEmpty(wrapper.getApplicantAddress().getCountry())) {
+            errors.add("If applicant resides outside of UK, Please provide the country of residence.");
+        }
+        YesOrNo respondentResideOutsideUK = wrapper.getRespondentResideOutsideUK();
+        if (ObjectUtils.isNotEmpty(respondentResideOutsideUK)
+            && respondentResideOutsideUK.equals(YesOrNo.YES)
+            && wrapper.getRespondentAddress() != null
+            && ObjectUtils.isEmpty(wrapper.getRespondentAddress().getCountry())) {
+            errors.add("If respondent resides outside of UK, Please provide the country of residence.");
+        }
+        return errors;
+    }
+
+    public List<String> validate(Map<String, Object> caseData) {
+        String applicantOutsideUK = String.valueOf(caseData.get(APPLICANT_RESIDE_OUTSIDE_UK));
+        Address applicantAddress = getAddress(caseData.get("applicantAddress"));
+
+        List<String> errors = new ArrayList<>();
+        if (applicantOutsideUK.equals("Yes")
+            && applicantAddress != null
+            && ObjectUtils.isEmpty(applicantAddress.getCountry())) {
+            errors.add("If applicant resides outside of UK, Please provide the country of residence.");
+        }
+
+        String respondentOutsideUK = String.valueOf(caseData.get(RESPONDENT_RESIDE_OUTSIDE_UK));
+        Address respondentAddress = getAddress(caseData.get("respondentAddress"));
+
+        if (respondentOutsideUK.equals("Yes")
+            && respondentAddress != null
+            && ObjectUtils.isEmpty(respondentAddress.getCountry())) {
+            errors.add("If respondent resides outside of UK, Please provide the country of residence.");
+        }
+        return errors;
+    }
+
+    /**
+     * Determines whether the recipient of a letter resides outside of the UK based on the given case data and recipient.
+     *
+     * @param caseData The case data as a map of key-value pairs.
+     * @param recipient The recipient of the letter.
+     * @return True if the recipient resides outside of the UK, false otherwise.
+     */
+    public boolean isRecipientResideOutsideOfUK(Map<String, Object> caseData, String recipient) {
+        String addressee = ObjectUtils.isNotEmpty(recipient) ?  WordUtils.capitalizeFully(recipient) : recipient;
```

Comment:
Is this check required? In what situation would there be no recipient?

Reply:
covered in test